src/ai/backend/manager/models/gql_models/group.py

```diff
@@ -217,13 +222,16 @@ async def get_node(cls, info: graphene.ResolveInfo, id) -> Self:
     async def get_connection(
         cls,
         info: graphene.ResolveInfo,
-        filter_expr: str | None = None,
-        order_expr: str | None = None,
-        offset: int | None = None,
-        after: str | None = None,
-        first: int | None = None,
-        before: str | None = None,
-        last: int | None = None,
+        scope: Optional[ScopeType] = None,
+        container_registry_scope: Optional[ContainerRegistryScope] = None,
```

Comment:
Just an idea. Since we are likely to have more orthogonal scopes mapped to project scopes, we could define a parameter type that contains `container_registry_scope` and use it here.
e.g.
```python
@dataclass # Can replace this with pydantic, TypedDict whatever you want!
class ProjectAdditionalScope:
    container_registry_scope: Optional[ContainerRegistryScope] = None

async def get_connection(
        cls,
        info: graphene.ResolveInfo,
        scope: Optional[ScopeType] = None,
        extra_scopes: ProjectAdditionalScope,
```

Reply:
I think the `ProjectAdditionalScope` name suggests that an additional scope is being added to the basic `ScopeType`.
However, in the current implementation, only one of `ScopeType` or `ContainerRegistryScope` is being used exclusively.
So, wouldn’t names like `ProjectAlternativeScope` or `ProjectSecondaryScope` be more appropriate? @fregataa

Reply:
https://github.com/lablup/backend.ai/blob/a10db5d776219690162140c2c2bb2faacf008dab/src/ai/backend/manager/models/rbac/__init__.py#L422-L43

Reply:
We decided to use ExtraScope.